### ISSUE TITLE
Citizen management flawed

### ISSUE THREAD

**Issue by vide0gamer** [LoneGazebo/Community-Patch-DLL#4007]
_1. Mod version (i.e Date - 4/23):_
2/27

_2. Mod list (if using Vox Populi only, leave blank):_

_3. Error description:_ 
I've noticed that often if I leave citizen management to default, some get assigned to clearly inferior (to me) tiles.

_4. Steps to reproduce (optional):_

---------------------------
Supporting information:
Please note that you can attach .zip files by dragging-and-dropping them. If possible, zip up all supporting data and post that way.

1. Log files (always attach your Logs folder, located at My Documents/My Games/Sid Meier's Civilization 5. Make sure you have enabled logging before experiencing an error! Go here to find out how: http://forums.civfanatics.com/showthread.php?t=487482):

[Logs.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/1772587/Logs.zip)

2. Save game (always attach a save that was made a turn before the error; located at My Documents/My Games/Sid Meier's Civilization 5/ModdedSaves):
[William_0015 BC-3400.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/1772589/William_0015.BC-3400.zip)

3. CvMiniDump.dmp file (attach if experiencing a game crash. Located at Program Files/Steam/steamapps/common/Sid Meier's Civilization V):

4. Screenshots (optional):
![civ5citman](https://user-images.githubusercontent.com/28330477/36866935-b8444d34-1d61-11e8-8e75-0a4e15511bae.png)
The hill is being used instead of the forested spices, which seems like a no brainer to me as far as this early in the game goes.

**Comment by LoneGazebo** [comment]
Cities value production a bit more when small to get early infrastructure up. You aren't in debt, so the gold doesn't matter as much, and your growth rate (6 turns) is acceptable to the AI.

G

**Comment by vide0gamer** [comment]
I guess I just look at it as getting 4 yield compared to 2.  I mean, if I work the spice, i'll hit growth in 4 turns instead of 6.  If I decide to work the hill with the new citizen, that means over 6 turns, i've traded 4 hammers for 12 food and 6 gold.  Is production really worth that much?
Also, if I switch to farming as my production, it stays the same, even though working the spice causes growth much faster than working the hill; this is probably a fringe case, as most of the time you would be under food focus anyway.

**Comment by vide0gamer** [comment]
Better example, later in the same game:
![civ5citman2](https://user-images.githubusercontent.com/28330477/36880004-7b97cd10-1d94-11e8-880e-b5ef0035c264.png)
[William_0043 BC-2280.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/1773433/William_0043.BC-2280.zip)
Why use the crab space instead of the atoll space?

**Comment by deadstarre** [comment]
both of these pictures look really bad. its a common theme that whenever Gov is making the wrong choice, its because he values some yield much more than a tile which has a bunch more of -other- yields. in the first case, he wants an extra hammer at a terrible exchange. in the 2nd pic, he wants the gold at a god awful exchange. cant he be taught as a general rule the tile with the most yields is the best one to work? because in most cases, thats true and then you dont end up with these.

**Comment by LoneGazebo** [comment]
I think it is just overemphasizing deficient yields. Going to keep tuning.

**Comment by LoneGazebo** [comment]
Mm, I also see what is causing perxterrain/feature yields to not be prioritized, I'm missing a +1 value.

G

**Comment by ilteroi** [comment]
in case this isn't fixed yet:

![image](https://user-images.githubusercontent.com/1358966/36951158-97e2f918-2000-11e8-8b52-cb40607970d5.png)

city is stuck at 1 pop ... why not work the jungle instead of the hill?

also, the tooltip says 17.2/15 food ... but the city is not growing.

**Comment by LoneGazebo** [comment]
I've been working on this.

**Comment by deadstarre** [comment]
instant bonuses can put the city over the bucket, but if the city is stagnant or starving the pop won't go up, it has to be a positive number to grow. occasionally annoying but i figured it had to be that way

**Comment by LoneGazebo** [comment]
@deadstarre yep

**Comment by vide0gamer** [comment]
Still seems to be some problems with this.
![civ5citman3](https://user-images.githubusercontent.com/28330477/37478639-2bd2e82e-2851-11e8-8f52-dbe237a83ba8.png)

[Pedro II_0064 BC-1440.zip](https://github.com/LoneGazebo/Community-Patch-DLL/files/1816355/Pedro.II_0064.BC-1440.zip)
Why forest instead of lake?

**Comment by LoneGazebo** [comment]
Herbalist bonus.

**Comment by vide0gamer** [comment]
That's still a 1/2 food instead of 1 food.  Lake is strictly better in this case.

**Comment by LoneGazebo** [comment]
You've got four forests being worked in total, so the game is getting you a +1f from the herbalist. They're worth the same in that case.

**Comment by vide0gamer** [comment]
...i'm only working 3 forests.

**Comment by LoneGazebo** [comment]
that plantation isn't on a forest? I can't really tell, looks like one to me.

**Comment by vide0gamer** [comment]
Plantation removes forest.

**Comment by LoneGazebo** [comment]
definitely, but on my phone it looked like it was still a forest. I'll look at the code.

**Comment by tu-79** [comment]
I've seen citizen manager ignoring lake tiles quite often, when they were the best option.

**Comment by LoneGazebo** [comment]
Yes it knows.

**Comment by tu-79** [comment]
That was not my question :)
I'm more interested in knowing which approach uses the algorithm. Incremental? Combinations? And how values are assigned to set priorities. I guess it is a big piece of code, but I'm willing to take a look. Maybe another pair of eyes? =)
